watson-developer-cloud/discovery-components
# chore: Add module path aliases

#### What do these changes do/fix?
This adds in module path aliases to replace the complex relative `../../` imports throughout `discovery-react-components`.

To get the aliases to work with Storybook, I had to include a different `baseUrl` in its `tsconfig.json` and use a `tsconfig-paths-webpack-plugin` in its Webpack config.

I also added in [`madge`](https://github.com/pahen/madge) to help us identify any circular dependencies. I added just the circular-dependencies check for now, but we could set up `madge` to do more in the future too.  I made this a separate command so we can run it ad hoc, and then it's also included in our lint.

I did identify 3 circular dependencies with `madge` that I also fixed in this PR.

related issue: https://github.ibm.com/Watson-Discovery/disco-widgets/issues/278

#### How do you test/verify these changes?
You can verify these changes by checking that builds, tests, Storybook, and the example app all still work as expected with no errors.

Note that you may need to restart VSCode to get it to recognize the module path aliases.

#### Have you documented your changes (if necessary)?
Yes, I've updated both the `README` and `CONTRIBUTING` with information about the module path aliases for imports and the new circular-dependencies command.

#### Are there any breaking changes included in this pull request?
These shouldn't be breaking changes.
